shell/platform/embedder/embedder.h

```diff
@@ -1205,6 +1209,19 @@ typedef struct {
   ///
   /// Embedders can provide either snapshot buffers or aot_data, but not both.
   FlutterEngineAOTData aot_data;
+
+  /// A callback that computes the locale the platform would natively resolve
+  /// to.
+  ///
+  /// The input parameter is an array of null terminated char* strings. Each
+  /// group of 3 strings represents one locale, with the strings supplying
```

Comment:
This seems unnecessarily error-prone; why not define a struct for a locale and make this an array of structs?

Reply:
Oh right, this is meant to be implemented by more than just internal code. I was initially passing it as a list of strings internally to avoid multiple back and forth conversions between differing locale definitions on platforms, but this surface should indeed be using the FlutterLocale struct.